site/content/posts/2021-06-7-Windows-Support.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Introducing Support for Windows Clusters
+image: /img/sonobuoy.svg
+excerpt: Sonobuoy can now run on Windows nodes meaning faster, consistent testing on all of your clusters.
+author_name: John Schnake
+author_url: https://github.com/vmware-tanzu
+categories: [kubernetes, sonobuoy, windows]
+# Tag should match author to drive author pages
+tags: ['Sonobuoy Team', 'John Schnake']
+date: 2021-06-07
+slug: sonobuoy-v0-51
+---
+
+Sonobuoy’s latest release, v0.51.0 introduces beta support for Windows, closing one of our longest standing roadmap items. This means that the Sonobuoy image can run on Windows nodes (including Windows versions "1809", "1903", "1909", "2004", and "20H2") in Kubernetes clusters as well as having a Windows compatible Sonobuoy client. This will simplify testing of Kubernetes clusters with Windows nodes and speed development of full Kubernetes support and certification of Windows clusters.
+
+In this blog post, I want to show you how to use Sonobuoy to run either the Kubernetes end-to-end tests or your own custom plugins on mixed node clusters. In a follow-up post, I will discuss specifically running the “e2e” plugin for the Kubernetes tests.
+
+## Why Sonobuoy Needs to Run on Windows Nodes
+
+A reasonable question would be why Sonobuoy needs to run on Windows nodes at all. If the cluster has some Linux nodes available, can the Sonobuoy pod just be placed there? Yes and no. First take a look at the general Sonobuoy architecture:
+
+![Architecture](/img/image1.png "Sonobuoy architecture")
+
+The aggregator pod can be placed anywhere in the cluster, so yes it can be placed on a Linux node if one is available. However, on any node where you want a plugin to run, Sonobuoy runs as a sidecar so that it can report progress and results. If you want to gather logs, execute custom tests, check OS settings, or gather debug information on Windows nodes, Sonobuoy needs to run on Windows.
+
+By supporting Windows, Sonobuoy enables you to use the same testing and debugging flow that you use for your existing Linux clusters. That means it is easier and faster to experiment with Windows cluster adoption.
+
+## Getting A Windows Cluster
+
+Obviously, if you would like to run Sonobuoy from a Windows node, you need to set up a Kubernetes cluster with Windows nodes. One of the easiest ways to get started is to use an Azure account and create an AKS cluster. It is simple but does require an account and costs fee based on resources used. Find more instructions [here](https://docs.microsoft.com/en-us/azure/aks/windows-container-cli).
```

Comment:
Remove "Obviously". Can change "simple" to "easy", and "fee" seems redundant here as you already have "costs".